internal/handlers/handlers.go

```diff
@@ -0,0 +1,95 @@
+package handlers
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/smtp"
+
+	"bitcoin_checker_api/config"
+	"bitcoin_checker_api/internal/models"
+	"bitcoin_checker_api/internal/repositories"
+
+	"github.com/gin-gonic/gin"
+)
+
+type Handler struct {
+	repository repositories.Repository
+	cfg        *config.Config
+}
+
+func NewHandler(cfg *config.Config, repository repositories.Repository) *Handler {
+	return &Handler{
+		cfg:        cfg,
+		repository: repository,
+	}
+}
+
+func rate(cfg *config.Config) (string, error) {
+	ctx := context.Background()
+	converter := models.NewConverter()
+	requestURL := fmt.Sprintf("%s%s", cfg.Converter.Endpoint, converter.GetQueryParams())
+	res, err := http.NewRequestWithContext(ctx, http.MethodGet, requestURL, nil)
+	if err != nil {
+		return "", err
+	}
+	body, _ := io.ReadAll(res.Body)
+	res.Body.Close()
+	return string(body), nil
+}
+
+func sendMail(email, data string) {
+	from := "from@gmail.com"
+	password := "<Email Password>"
+
+	// Receiver email address.
+	to := []string{
+		email,
+	}
+
+	// smtp server configuration.
+	smtpHost := "smtp.gmail.com"
+	smtpPort := "587"
+
+	// Message.
+	message := []byte(data)
+
+	// Authentication.
+	auth := smtp.PlainAuth("", from, password, smtpHost)
+
+	// Sending email.
+	err := smtp.SendMail(smtpHost+":"+smtpPort, auth, from, to, message)
+	if err != nil {
+		fmt.Printf("Error: %s", err) //nolint:forbidigo // log for now
+		return
```

Comment:
If you want to print an error it's better to use `fmt.Fprintf(os.Stderr, ...)` like this:
```suggestion
                fmt.Fprintf(os.Stderr, "sending an email: %v", err)
		return
```